kong/db/dao/init.lua

```diff
@@ -328,12 +334,18 @@ local function check_update(self, key, entity, options, name)
     return nil, err, err_t
   end
 
-  local ok, errors = self.schema:validate_update(entity_to_update)
+  local ok, errors = self.schema:validate_update(entity_to_update, entity)
   if not ok then
     local err_t = self.errors:schema_violation(errors)
     return nil, nil, tostring(err_t), err_t
   end
 
+  entity_to_update, err = self.schema:transform(entity_to_update, entity)
```

Comment:
It seems that transforms happen immediately on `insert` and `update`, but never on `read`. Should we have an `on` = insert | update | read field in transforms?

Reply:
`on = read, write, insert, update, upsert...` OR just `on = read, write`. Perhaps... I don't know. This does not implement read-transformations, and doesn't differentiate on write operations at this point. But I guess same comment applies here, why not just pass `context` to transformation function and let the function decide. It reminds a lot of that old `self_check` that we used to have, e.g. we could call it `mutate_and_or_validate` and here is the whole entity and a context.